FlowCrypt/Functionality/Services/Key Services/PassPhraseStorage.swift

```diff
@@ -0,0 +1,107 @@
+//
+//  PassPhraseStorageService.swift
+//  FlowCrypt
+//
+//  Created by Anton Kharchevskyi on 02.06.2021.
+//  Copyright © 2021 FlowCrypt Limited. All rights reserved.
+//
+
+import UIKit
+
+protocol PassPhraseStorageType {
+    func getPassPhrases() -> [PassPhrase]
+    func savePassPhrase(with passPhrase: PassPhrase, inStorage: Bool)
+    func updatePassPhrase(with passPhrase: PassPhrase, inStorage: Bool)
+}
+
+protocol EmailProviderType {
+    var email: String? { get }
+}
+
+final class PassPhraseStorage: PassPhraseStorageType {
+    private lazy var logger = Logger.nested(Self.self)
+
+    let currentUserEmail: String?
+    let storage: EncryptedPassPhraseStorage
+    let localStorage: LocalPassPhraseStorageType
+    let timeoutInSeconds: Int
```

Comment:
There should be a better separation of responsibilities. Everything relating to time tracking and expiring old in-memory passphrases should be handled in `InMemoryPassPhraseStorage`. This class that combines the two methods should be more dumb. I'll file a separate issue.

Reply:
To handle in future PR here #363